### PR TITLE
Hide sensitive information from pod logs.

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
Hide sensitive information from pod logs.

Signed-off-by: Oleksandr Andriienko <oandriie@redhat.com>

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/ok-to-test

File: cmd/api/main.go [hunk 1]
```diff
@@ -87,10 +88,9 @@ func main() {
 
 	dbURI := fmt.Sprintf("host=%s user=%s password=%s dbname=%s port=%s sslmode=%s", configFile.DB_HOST, configFile.DB_USER, configFile.DB_PASSWORD, configFile.DB_NAME, configFile.DB_PORT, configFile.DB_SSLMODE)
 
-	db, err := gorm.Open(postgres.Open(dbURI), &gorm.Config{})
-	if err != nil {
-		log.Fatalf("Failed to open the results.db: %v", err)
-	}
+	db, err := gorm.Open(postgres.Open(dbURI), &gorm.Config{
+		Logger: logger.Default.LogMode(logger.Silent),
+	})
```
Comment (adambkaplan) [inline cmd/api/main.go:93]: Does this mean that the GORM logs are permanently suppressed? If yes, can we use the zap log level to turn these back on? For example - log if "debug" is set.

Also Need to ensure that we catch the error opening the database connection and fail:

```suggestion
	db, err := gorm.Open(postgres.Open(dbURI), &gorm.Config{
		Logger: logger.Default.LogMode(logger.Silent),
	})
	if err != nil {
		log.Fatalf("Failed to open the results.db: %v", err)
	}
```

File: test/e2e/01-install.sh [hunk 2]
```diff
@@ -26,7 +26,7 @@ kubectl apply --filename ${TEKTON_PIPELINE_CONFIG}
 
 echo "Generating DB secret..."
 # Don't fail if the secret isn't created - this can happen if the secret already exists.
-kubectl create secret generic tekton-results-postgres --namespace="tekton-pipelines" --from-literal=POSTGRES_USER=postgres --from-literal=POSTGRES_PASSWORD=$(openssl rand -base64 20) || true
+kubectl create secret generic tekton-results-postgres --namespace="tekton-pipelines" --from-literal=POSTGRESQL_USER=postgres --from-literal=POSTGRESQL_PASSWORD=$(openssl rand -base64 20) || true
```
Comment (adambkaplan) [inline test/e2e/01-install.sh:29]: Will this break our e2e tests?

**Review by adambkaplan — CHANGES_REQUESTED** [review]

File: test/e2e/01-install.sh (hunk 2 above)
Comment (adambkaplan) [reply test/e2e/01-install.sh:29]: looks like this will undo #270 - please revert this change.

File: cmd/api/main.go (hunk 1 above)
Comment (adambkaplan) [reply cmd/api/main.go:93]: @AndrienkoAleksandr following up on our slack conversation - I think for now we can suppress these log messages. I will create a follow up issue to make these logs configurable.

Comment (AndrienkoAleksandr) [reply cmd/api/main.go]: Hello, default log library doesn't have log level feature, so if you don't mind I updated my pr to use zap logger and I made log level configurable using config file...

Comment (adambkaplan) [reply cmd/api/main.go]: That's fine, though I suspect we will need to propagate this logger to other spots in the code.

File: cmd/api/main.go [hunk 3]
```diff
@@ -87,7 +92,11 @@ func main() {
 
 	dbURI := fmt.Sprintf("host=%s user=%s password=%s dbname=%s port=%s sslmode=%s", configFile.DB_HOST, configFile.DB_USER, configFile.DB_PASSWORD, configFile.DB_NAME, configFile.DB_PORT, configFile.DB_SSLMODE)
 
-	db, err := gorm.Open(postgres.Open(dbURI), &gorm.Config{})
+	gormConf := &gorm.Config{}
+	if logConf.Level.Level() == zap.DebugLevel {
+		gormConf.Logger = logger.Default.LogMode(logger.Silent)
+	}
```
Comment (adambkaplan) [inline cmd/api/main.go:98]: I think this does the opposite of what we want. Logging higher than "debug" should silence the gorm logger.

**Comment by adambkaplan** [comment]
> the code looks good - just a quick question - what is the sensitive information that we are hiding here? Is that the DB/GORM logs?

Mainly the DB/GORM logs. A few other log messages are also being downgraded from "Info" to "Debug" level.

In addition, #275 will add additional observability in the logs.

**Comment by adambkaplan** [comment]
@AndrienkoAleksandr looks good. Please squash your commits and I can lgtm.

**Comment by AndrienkoAleksandr** [comment]
> @AndrienkoAleksandr looks good. Please squash your commits and I can lgtm.

Sorry, for delay, done.

**Review by khrm — COMMENTED** [review]
/kind feature
We need this label for release as discussed in the last WG call.

**Pull request merged**